core/bootstrap/src/main/java/org/wildfly/swarm/bootstrap/modules/ApplicationModuleFinder.java

```diff
@@ -121,7 +120,7 @@ protected void addAsset(ModuleSpec.Builder builder, ApplicationEnvironment env)
             name = name.substring(0, dotLoc);
         }
 
-        File tmp = TempFileManager.INSTANCE.newTempFile(name, ext);
+        File tmp = File.createTempFile(name, ext);
```

Comment:
Typically we change it the other way round. Why can't we keep this using the `TempFileManager` and let the shutdown hooks registered below to only close the `ResourceLoader`s? Maybe we have to make sure that the `ResourceLoader`s are closed _before_ we delete the file?